TourMate/TourMate/Frontend/Plan/Views/AddPlanView/AddAccommodationView.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  AddAccommodationView.swift
+//  TourMate
+//
+//  Created by Tan Rui Quan on 9/4/22.
+//
+
+import SwiftUI
+
+struct AddAccommodationView: View {
+    @Environment(\.dismiss) var dismiss
+    @StateObject var viewModel: AddAccommodationViewModel
+    var dismissAddPlanView: DismissAction
+
+    var planFormView: some View {
+        PlanFormView(viewModel: viewModel) {
+            Section("Date & Time") {
+                DatePicker("Check-in Date",
+                           selection: $viewModel.planStartDate,
+                           in: viewModel.lowerBoundDate...viewModel.upperBoundDate,
+                           displayedComponents: [.date, .hourAndMinute])
+
+                DatePicker("Check-out Date",
+                           selection: $viewModel.planEndDate,
+                           in: viewModel.lowerBoundDate...viewModel.upperBoundDate,
+                           displayedComponents: [.date, .hourAndMinute])
+            }
+
+            Section("Location") {
+                AddressTextField(title: "Address", location: $viewModel.location)
+            }
+        }
+    }
+
+    var body: some View {
+        NavigationView {
+            Group {
+                if viewModel.hasError {
+                    Text("Error Occurred")
+                } else if viewModel.isLoading {
+                    ProgressView()
+                } else {
+                    planFormView
+                }
+            }
+            .navigationTitle("New Accommodation")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Add") {
+                        Task {
+                            await viewModel.addAccommodation()
+                            dismiss()
+                            dismissAddPlanView()
+                        }
+                    }
+                    .disabled(!viewModel.canSubmitPlan || viewModel.isLoading || viewModel.hasError)
+                }
+
+                ToolbarItem(placement: .cancellationAction) {
+                    Button("Cancel", role: .destructive) {
+                        dismiss()
+                    }
+                    .disabled(viewModel.isLoading)
+                }
+            }
+        }
+    }
+}
```

Comment:
I think these can be abstracted, since the viewmodels are already following an inheritance structure i think these can as well.

Reply:
The viewModel should all have the addPlan() method which overrides the PlanViewModel one